### PR TITLE
fix #308694 fixed open files with the path name that contains non-ASC…

### PR DESCRIPTION
**Pull request by igorkorsukov**
Resolves: https://musescore.org/en/node/308694

### CODE REVIEW

File: mscore/exportaudio.cpp [hunk 1]
```diff
@@ -254,7 +257,12 @@ bool MuseScore::saveAudio(Score* score, const QString& name)
                   if ((mode & QIODevice::WriteOnly) == 0) {
                         return false;
                         }
-                  sf     = sf_open(qPrintable(filename), SFM_WRITE, &info);
+
+                  file.setFileName(filename);
+                  if (!file.open(mode)) {
+                      return false;
```
Comment (Jojo-Schmitz) [inline mscore/exportaudio.cpp:263]: old code style, 6 spaces and banner style indent ;-)

File: mscore/exportaudio.cpp [hunk 2]
```diff
@@ -254,7 +257,12 @@ bool MuseScore::saveAudio(Score* score, const QString& name)
                   if ((mode & QIODevice::WriteOnly) == 0) {
                         return false;
                         }
-                  sf     = sf_open(qPrintable(filename), SFM_WRITE, &info);
```
Comment (Jojo-Schmitz) [inline mscore/exportaudio.cpp:257]: Actually I have the suspicion that this `qPrintable()` is the culprit already and alone.

**Review by Jojo-Schmitz — APPROVED** [review]

File: mscore/exportaudio.cpp (hunk 2 above)
Comment (igorkorsukov) [reply mscore/exportaudio.cpp:257]: At first, I thought so too, replaced it with qUtf8Printable - not working.
Started searching, found that the problem is known https://github.com/erikd/libsndfile/issues/393
Especially for Windows, there is a separate function for opening files (need to do `#ifdef WINDOWS ...`)
I looked at what other opening functions are there, I saw that there is a function that takes a file descriptor and a decision came to my mind that I made, I think it is the most simple and reliable.

Comment (Jojo-Schmitz) [reply mscore/exportaudio.cpp:257]: It certainly looks sweet and simple.

File: mscore/exportaudio.cpp (hunk 1 above)
Comment (igorkorsukov) [reply mscore/exportaudio.cpp]: fixed :)

**Pull request merged**